nixos/modules/services/development/livebook.nix

```diff
@@ -0,0 +1,101 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.livebook;
+in
+{
+  options.services.livebook = {
+    enableUserService = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to enable a user service for Livebook.  This implies
+        `installUserService`.
+      '';
+    };
+
+    installUserService = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to install a user service for Livebook.  Unless
+        `enableUserService` is also set, users will have to manually activate
+        the service with `systemctl --user enable livebook`.
+      '';
+    };
+
+    environment = mkOption {
+      type = with types; attrsOf str;
+      description = lib.mdDoc ''
+        The environment to pass to the Livebook service.  You need to set at least
+        `LIVEBOOK_PASSWORD`.  See `livebook server --help` for other options.
+      '';
+      example = {
+        LIVEBOOK_PASSWORD = "a password at least 30 characters long";
+        LIVEBOOK_PORT = 30123;
+      };
+    };
```

Comment:
Could we make this take an `environmentFile` instead? It makes it a lot easier to provide secrets without putting the secrets in the nix store.

Reply:
Done.